options/rtld/generic/linker.cpp

```diff
@@ -558,12 +558,12 @@ frg::expected<LinkerError, void> ObjectRepository::_fetchFromFile(SharedObject *
 			#if MLIBC_MAP_DSO_SEGMENTS
 				void *map_pointer;
 				if(mlibc::sys_vm_map(reinterpret_cast<void *>(map_address),
-						backed_map_size, prot | PROT_WRITE,
+						backed_map_size, prot,
 						MAP_PRIVATE | MAP_FIXED, fd, phdr->p_offset - misalign, &map_pointer))
 					__ensure(!"sys_vm_map failed");
 				if(total_map_size > backed_map_size)
 					if(mlibc::sys_vm_map(reinterpret_cast<void *>(map_address + backed_map_size),
-							total_map_size - backed_map_size, prot | PROT_WRITE,
+							total_map_size - backed_map_size, prot,
```

Comment:
AFAICT this was done to ensure that the memset() below succeeds.

Reply:
Addressed.